lib/BackgroundJob/RetentionJob.php

```diff
@@ -187,8 +187,15 @@ private function checkFileId($fileid) {
 	 */
 	private function expireNode(Node $node, \DateTime $deleteBefore) {
 		$mtime = new \DateTime();
+
+		// Fallback is the mtime
 		$mtime->setTimestamp($node->getMTime());
 
+		// Use the upload time if we have it
+		if ($node->getUploadTime() !== 0) {
```

Comment:
what if it actually was uploaded January 1st 1970? :stuck_out_tongue_closed_eyes: